packages/@react-spectrum/tag/src/Tag.tsx

```diff
@@ -10,82 +10,75 @@
+  let {isFocusVisible, focusProps} = useFocusRing({within: true});
   let {clearButtonProps, labelProps, tagProps} = useTag({
     ...props,
-    isRemovable: removable,
-    isDisabled: disabled,
-    validationState: validationState !== undefined ? validationState : groupValidationState,
-    onRemove: props.onRemove || onRemove,
-    role
-  });
-  let {role: buttonRole, ...otherButtonProps} = clearButtonProps;
-  let icon = props.icon || (isInvalid && <Alert />);
+    isRemovable,
+    isDisabled,
+    item,
+    onRemove
+  }, state);
 
   return (
-    <FocusRing focusRingClass={classNames(styles, 'focus-ring')}>
-      <div
-        {...styleProps}
-        {...mergeProps(tagProps, hoverProps)}
-        className={classNames(
-          styles,
-          'spectrum-Tags-item',
-          {
-            'is-disabled': disabled,
-            // 'is-selected': isSelected,
-            'spectrum-Tags-item--removable': removable,
-            'is-invalid': isInvalid,
-            'is-hovered': isHovered
-          },
-          styleProps.className
-        )}>
-        {icon && React.cloneElement(icon, {size: 'S', UNSAFE_className: classNames(styles, 'spectrum-Tags-itemIcon')})}
-        <span
-          {...labelProps}
-          className={classNames(styles, 'spectrum-Tags-itemLabel')}>
-          {props.children}
-        </span>
-        {removable &&
-          <span role={buttonRole}>
-            <ClearButton
-              tabIndex={tagProps.tabIndex}
-              focusClassName={classNames(styles, 'is-focused')}
-              UNSAFE_className={classNames(styles, 'spectrum-Tags-itemClearButton')}
-              {...otherButtonProps} />
-          </span>
+    <div
+      {...mergeProps(tagProps, hoverProps, focusProps, labelProps, otherProps)}
+      role={tagProps.role}
+      className={classNames(
+        styles,
+        'spectrum-Tags-item',
+        {
+          'is-disabled': isDisabled,
+          'focus-ring': isFocusVisible,
+          'is-focused': isFocusVisible,
+          'spectrum-Tags-item--removable': isRemovable,
+          'is-hovered': isHovered
         }
-      </div>
-    </FocusRing>
+      )}>
+      <SlotProvider
+        slots={{
+          icon: {UNSAFE_className: classNames(styles, 'spectrum-Tag-icon'), size: 'XS'},
+          text: {UNSAFE_className: classNames(styles, 'spectrum-Tag-content')}
+        }}>
+        {typeof props.children === 'string' ? <Text><span role="gridcell">{props.children}</span></Text> : props.children}
+        {isRemovable && <TagRemoveButton item={item} {...clearButtonProps} UNSAFE_className={classNames(styles, 'spectrum-Tag-action')} />}
+      </SlotProvider>
+    </div>
+  );
+}
+
+function TagRemoveButton(props) {
+  props = useSlotProps(props, 'tagRemoveButton');
+  let {styleProps} = useStyleProps(props);
+  let clearBtnRef = useRef();
+
+  return (
+    <span
+      {...styleProps}
+      ref={clearBtnRef}>
```

Comment:
This `span` should have `role="gridcell"`, which should either come from `clearButtonProps` or be added as a static prop here, as we seem to be doing at Line 63 above.